packages/rocketchat-lib/server/lib/sendNotificationsOnMessage.js

```diff
@@ -24,6 +24,10 @@ function replaceMentionedUsernamesWithFullNames(message, mentions) {
 	return message;
 }
 
+function canSendMessageToRoom(room, username) {
+	return !((room.muted || []).includes(username));
```

Comment:
Not sure I got the logic here... the username needs to be in the `mentions` to be able to reply? 🤔

Reply:
Where did you see `mentions` @rafaelks ?

Reply:
Here: https://github.com/RocketChat/Rocket.Chat/pull/8905/files#diff-819f1684d8dba4a649500c1ce5166000R144

Reply:
@rafaelks The code is a little bit confusing, but it is the array of users that will receive the push, not the mentioned ones only.

Reply:
OK... and what if the room is read only? This code isn't checking that AFAIK.

Reply:
actually this is the code that checks it.. readonly rooms have a list of `muted` users..

Reply:
@sampaiodiego So a `readonly` room  will have all usernames in `muted` array?

Reply:
yep.. not a good implementation I know, but this is how it works now :man_shrugging: